src/common/offline_transformations/src/pruning/init_masks.cpp

```diff
@@ -75,8 +76,34 @@ class ngraph::pass::init_masks::InitMatMulMask : public MatcherPass {
             NodeVector weights_calculation_nodes;
             auto cur_node = matmul->get_input_node_shared_ptr(1);
 
+            if (!cur_node->output(0).get_partial_shape().is_static()) return false;
+            auto dim_order = std::vector<int64_t>();
+            const auto input_size = cur_node->get_shape().size();
```

Comment:
here you assume `curr_node` has single output. can we have an assert about it? or adopt algo to work with all outputs?

Reply:
We just need an output shape. I fix that but additional unit test is needed